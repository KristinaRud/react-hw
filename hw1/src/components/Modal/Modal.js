import { Component } from "react";
import Button from "../Button/Button";
import "./Modal.scss";

class Modal extends Component {
	render() {
		const { header, closeButton, text, actions, closeModal } = this.props;
		return (
			<div className="modal-wrapper" onClick={closeModal}>
				<div className="modal" onClick={(e) => e.stopPropagation()}>
					<div className="modal-box">
						<div className="modal-header">
							<h4>{header}</h4>
							{closeButton && (
								<Button
									className="modal-close"
									onClick={closeModal}
									text={
										<svg
											viewBox="0 0 16 16"
											width="16"
											height="16"
										>
											<path fill="#FFF" d="m9.3 8 6.1-6.1c.4-.4.4-.9 0-1.3s-.9-.4-1.3 0L8 6.7 1.9.6C1.6.3 1 .3.6.6c-.3.4-.3 1 0 1.3L6.7 8 .6 14.1c-.4.4-.3.9 0 1.3l.1.1c.4.3.9.2 1.2-.1L8 9.3l6.1 6.1c.4.4.9.4 1.3 0s.4-.9 0-1.3L9.3 8z" />
										</svg>
									}
								/>
							)}
						</div>
						<div className="modal-content">
							<div className="input">{text}</div>
						</div>
						<div className="modal-footer">
							<div className="button-wrapper">{actions}</div>
						</div>
					</div>
				</div>
			</div>
		);
	}
}

export default Modal;
