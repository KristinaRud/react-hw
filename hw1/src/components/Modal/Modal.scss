.modal-wrapper {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 100;
    background-color: rgba(0, 0, 0, 0.4);
}

.modal {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    max-width: 560px;
    width: 100%;
    box-sizing: border-box;
    background-color: RGBA(231, 76, 60, 1);
    border-radius: 6px;

    .modal-close {
        position: absolute;
        right: 16px;
        top: 16px;
        display: inline-flex;
        vertical-align: middle;
        text-align: center;
        box-sizing: border-box;
        align-items: center;
        justify-content: center;
        text-decoration: none !important;
        outline: none;
        -webkit-appearance: none;
        border: none;
        cursor: pointer;
        user-select: none;
        background-color: transparent;
    }

    .modal-header {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        padding: 30px;
        background-color: RGBA(212, 70, 55, 1);
        margin-bottom: 24px;
        font-size: 22px;
        font-weight: 500;
        font-style: normal;
        text-align: left;
        color: #FFFFFF;
        border-radius: 6px;
    }

    h4 {
        margin: 0;
    }

    .modal-content {
        padding: 40px;
        font-size: 15px;
        font-weight: 400;
        font-style: normal;
        text-align: center;
        color: #FFFFFF;
    }

    .modal-footer {
        padding-bottom: 30px;
    }

    .button-wrapper {
        display: flex;
        justify-content: center;
        align-items: center;
        gap: 10px;
    }
}

.btn {
    position: relative;
    display: inline-flex;
    vertical-align: middle;
    text-align: center;
    box-sizing: border-box;
    align-items: center;
    justify-content: center;
    text-decoration: none !important;
    outline: none;
    -webkit-appearance: none;
    border: none;
    cursor: pointer;
    user-select: none;
    background-color: #6b78d6;
    font-family: Roboto, sans-serif;
    font-weight: 400;
    font-size: 16px;
    color: #fff;
    min-width: 101px;
	height: 41px;
    border-radius: 4px;
}

.btn:has(svg){
    background: none;
    min-width: 20px;
    height: 20px;
    color: #FFFFFF;
    width: auto;
    height:auto;
}