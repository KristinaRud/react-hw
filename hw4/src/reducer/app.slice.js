import { createSlice } from "@reduxjs/toolkit";



const initialState = {
	currentComics: [],
	// profile: {
	// 	name: "John",
	// 	lastName: "Doe",
	// 	email: "John@gmail.com",
	// },
	isModal: false,
}
const appSlice = createSlice({
	name: "app",
	initialState,
	reducers: {
		actionModal: (state, { payload }) => {
			state.isModal = payload;
		},
		actionCurrentComics: (state, { payload }) => {
			state.currentComics = { ...payload };
		},
	}
})

export const { actionModal, actionCurrentComics } = appSlice.actions
export default appSlice.reducer
