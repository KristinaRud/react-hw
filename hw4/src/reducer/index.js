import appReducer, { actionCurrentComics, actionModal } from "./app.slice"
//import tvReducer, {actionFetchPageTv, actionFetchSliderTv} from "./tv.slice"
import comicsReducer, { actionFetchSliderComics } from "./comics.slice"
import favoriteReducer, { actionSetFavorite, actionToggleFavorite } from "./favorite.slice"
import orderReducer, { actionSetOrder, actionAddOrder, actionIncrementCount, actionDecrementCount, actionDeleteOrderItem } from "./order.slice"

export {
	appReducer, actionCurrentComics, actionModal,
	comicsReducer, actionFetchSliderComics,
	favoriteReducer, actionSetFavorite, actionToggleFavorite,
	orderReducer, actionSetOrder, actionAddOrder, actionIncrementCount, actionDecrementCount, actionDeleteOrderItem
}
