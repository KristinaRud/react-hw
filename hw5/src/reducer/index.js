import appReducer, { actionCurrentComics, actionModal } from "./app.slice"
import comicsReducer, { actionFetchSliderComics } from "./comics.slice"
import favoriteReducer, { actionSetFavorite, actionToggleFavorite } from "./favorite.slice"
import orderReducer, { actionSetOrder, actionAddOrder, actionIncrementCount, actionDecrementCount, actionDeleteOrderItem } from "./order.slice"
import formReducer, {actionForm, actionConfirm} from "./form.slice"

export {
	appReducer, actionCurrentComics, actionModal,
	comicsReducer, actionFetchSliderComics,
	favoriteReducer, actionSetFavorite, actionToggleFavorite,
	orderReducer, actionSetOrder, actionAddOrder, actionIncrementCount, actionDecrementCount, actionDeleteOrderItem,
	formReducer, actionForm, actionConfirm
}
