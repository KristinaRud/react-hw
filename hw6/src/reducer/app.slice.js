import { createSlice } from "@reduxjs/toolkit";



const initialState = {
	currentComics: [],
	isModal: false,
};
const appSlice = createSlice({
	name: "app",
	initialState,
	reducers: {
		actionModal: (state, { payload }) => {
			state.isModal = payload;
		},
		actionCurrentComics: (state, { payload }) => {
			state.currentComics = { ...payload };
		},
	}
})

export const { actionModal, actionCurrentComics } = appSlice.actions
export default appSlice.reducer
