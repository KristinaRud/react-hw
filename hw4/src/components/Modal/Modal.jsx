import "./Modal.scss";
import PropTypes from "prop-types";
import Button from "../Button/Button";

const Modal = ({modalTitle, closeModal, handlerModal, content, buttonContent}) => {
  const { title, img, creators, price } = content;
  return (
    <div className="modal-wrapper" onClick={closeModal}>
      <div className="modal" onClick={(e) => e.stopPropagation()}>
        <div className="detail-item-bg">
          <img className="back-img" src={img.url + img.background} />
        </div>
        <div className="modal-box">
          {/* надо поменять на компонент button */}
          <button type="button" onClick={closeModal} className="modal-close">
            <svg viewBox="0 0 16 16" width="16" height="16">
              <path
                fill="#E62429"
                d="m9.3 8 6.1-6.1c.4-.4.4-.9 0-1.3s-.9-.4-1.3 0L8 6.7 1.9.6C1.6.3 1 .3.6.6c-.3.4-.3 1 0 1.3L6.7 8 .6 14.1c-.4.4-.3.9 0 1.3l.1.1c.4.3.9.2 1.2-.1L8 9.3l6.1 6.1c.4.4.9.4 1.3 0s.4-.9 0-1.3L9.3 8z"
              />
            </svg>
          </button>
          {modalTitle}
          <div className="modal-header">
            <h4>{title}</h4>
          </div>
          <div className="modal-content">
            <div>
              <img
                className="item-img"
                alt={title}
                title={title}
                src={img.url + img.clean}
              />
            </div>
            <div>
              <h3>Creators</h3>
              <p>{creators}</p>
            </div>
          </div>
        </div>
        <div className="modal-footer">
          <div className="button-wrapper">
            <Button
              className="btn-buy"
              onClick={() => {
                handlerModal();
                closeModal();
              }}
              content={ buttonContent!=="Delete"? <span className="innerFill">Buy ${price}</span> : <span className="innerFill">{buttonContent}</span>}
            />
            {/* <button className="btn" onClick={closeModal} type="button">Cancel</button> */}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Modal;

Modal.propTypes = {
  closeModal: PropTypes.func.isRequired,
  handlerModal: PropTypes.func.isRequired,
  content: PropTypes.shape({
    title: PropTypes.string.isRequired,
    img: PropTypes.shape({
      url: PropTypes.string,
      portrait_uncanny: PropTypes.string,
      background: PropTypes.string,
      clean: PropTypes.string,
    }).isRequired,
    creators: PropTypes.string,
    price: PropTypes.string,
  }).isRequired,
};
