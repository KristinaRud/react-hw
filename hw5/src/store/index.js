import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import thunk from "redux-thunk";
import { comicsReducer, appReducer, orderReducer, favoriteReducer} from "../reducer";




const store = configureStore({
	reducer: {
		order: orderReducer,
		favorite: favoriteReducer,
		comics: comicsReducer,
		app: appReducer,
	},
	middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger, thunk),
	devTools: true,
});
export default store;
