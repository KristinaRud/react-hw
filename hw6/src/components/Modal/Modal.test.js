import Modal from "./Modal";
import {render, screen, fireEvent} from "@testing-library/react";
import { Provider, useDispatch } from 'react-redux';
import store from "../../store";
import '@testing-library/jest-dom';
import {actionModal} from "../../reducer";

const Component = ({modalTitle,buttonContent})=>{
    const dispatch=useDispatch();
    return(
        <>
            <button dispatch={()=>actionModal(false)}>CLOSE</button>
            <button onClick={()=>{dispatch(actionModal(true))}}>OPEN</button>
            <Modal modalTitle={modalTitle} buttonContent={buttonContent}/>
        </>
    )
}

const MockedProvider=({modalTitle, buttonContent})=>(
    <Provider store={store}>
        <Component modalTitle={modalTitle} buttonContent={buttonContent} />
    </Provider>
)

describe('Modal snapshot testing', ()=>{
    test('should Modal match snapshot', () => {
        const {asFragment} = render(<MockedProvider modalTitle={<h2>Test</h2>} buttonContent="Button"/>);
        fireEvent.click(screen.getByText('OPEN'));
        expect(asFragment()).toMatchSnapshot();
    })
})

describe('Modal close testing background', ()=>{
    test('should Modal background close work', () => {
        const {container}=render(<MockedProvider modalTitle={<h2>Test</h2>} buttonContent="Button"/>);
        
        fireEvent.click(screen.getByText("OPEN"));

        expect(screen.queryByText('Test')).toBeInTheDocument();
        
        fireEvent.click(container.querySelector('.modal-wrapper'));

        expect(screen.queryByText("<h2>Test</h2>")).not.toBeInTheDocument();
    })
})

describe('Modal close testing icon', ()=>{
    test('should Modal icon close work', () => {
        const {container}=render(<MockedProvider modalTitle={<h2>Test</h2>} buttonContent="Button"/>);

        fireEvent.click(screen.getByText('OPEN'));

        expect(container.querySelector('.modal-close')).toBeInTheDocument();

        fireEvent.click(container.querySelector('.modal-close'));

        expect(screen.queryByText("<h2>Test</h2>")).not.toBeInTheDocument();
    })
})