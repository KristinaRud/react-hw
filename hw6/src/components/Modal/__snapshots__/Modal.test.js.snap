// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot testing should Modal match snapshot 1`] = `
<DocumentFragment>
  <button>
    CLOSE
  </button>
  <button>
    OPEN
  </button>
  <div
    class="modal-wrapper"
  >
    <div
      class="modal"
    >
      <div
        class="detail-item-bg"
      >
        <img
          class="back-img"
          src="NaN"
        />
      </div>
      <div
        class="modal-box"
      >
        <button
          class="modal-close"
          type="button"
        >
          <svg
            height="16"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="m9.3 8 6.1-6.1c.4-.4.4-.9 0-1.3s-.9-.4-1.3 0L8 6.7 1.9.6C1.6.3 1 .3.6.6c-.3.4-.3 1 0 1.3L6.7 8 .6 14.1c-.4.4-.3.9 0 1.3l.1.1c.4.3.9.2 1.2-.1L8 9.3l6.1 6.1c.4.4.9.4 1.3 0s.4-.9 0-1.3L9.3 8z"
              fill="#E62429"
            />
          </svg>
        </button>
        <div
          class="modal-title"
        >
          <h2>
            Test
          </h2>
        </div>
        <div
          class="modal-header"
        >
          <h4 />
        </div>
        <div
          class="modal-content"
        >
          <div>
            <img
              class="item-img"
              src="NaN"
            />
          </div>
          <div>
            <h3>
              Creators
            </h3>
            <p />
          </div>
        </div>
      </div>
      <div
        class="modal-footer"
      >
        <div
          class="button-wrapper"
        >
          <button
            class="btn-buy"
            type="button"
          >
            <span
              class="innerFill"
            >
              Buy $
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
